packages/controller-utils/src/util.ts

```diff
@@ -16,6 +16,25 @@ import { MAX_SAFE_CHAIN_ID } from './constants';
 
 const TIMEOUT_ERROR = new Error('timeout');
 
+const PROTOTYPE_POLLUTION_BLOCKLIST = [
+  '__proto__',
+  'constructor',
+  'prototype',
+] as const;
+
+/**
+ * Checks whether a dynamic property key could be used in
+ * a [prototype pollution attack](https://portswigger.net/web-security/prototype-pollution).
+ *
+ * @param key - The dynamic key to validate.
+ * @returns Whether the given dynamic key is safe to use.
+ */
+export function isSafeDynamicKey(key: string): boolean {
```

Comment:
Should we write a test for this function too?

Reply:
Tests added! https://github.com/MetaMask/core/pull/4041/commits/03333afd8517922f14cf0da4dc42b06f45e1991c